prism/src/main/java/org/apache/falcon/resource/AbstractInstanceManager.java

```diff
@@ -676,6 +683,79 @@ public TriageResult triageInstance(String entityType, String entityName, String
         }
     }
 
+    public InstancesResult searchInstances(String type, String nameSubsequence, String tagKeywords,
```

Comment:
It looks like that searchInstances function is doing multiple things. Is it possible to split the function to reflect more appropriately.